src/game_interpreter.cpp

```diff
@@ -4643,6 +4645,29 @@ bool Game_Interpreter::CommandManiacCallCommand(lcf::rpg::EventCommand const&) {
 	return true;
 }
 
+bool Game_Interpreter::CommandSetFlag(lcf::rpg::EventCommand const& com) {
+
+	std::string flagName = Utils::LowerCase(ToString(com.string));
+	int flagValue = ValueOrVariable(com.parameters[0], com.parameters[1]);
+
+	if (flagName == "rpg2k3commands") //2k3 commands on 2k games
+		Player::game_config.patch_rpg2k3_commands.Set(flagValue);
+	if (flagName == "dynrpg") // dynrpg patch
+		Player::game_config.patch_dynrpg.Set(flagValue);
+	if (flagName == "maniacs" || flagName == "maniac") // maniacs patch
+		Player::game_config.patch_maniac.Set(flagValue);
+	if (flagName == "keypatch") // ineluki's key patch
+		Player::game_config.patch_key_patch.Set(flagValue);
+	if (flagName == "unlockpics") // unlock calling while dialogs box are open
+		Player::game_config.patch_unlock_pics.Set(flagValue);
+	if (flagName == "commonthisevent") //common events can use "this event" as a parameter
+		Player::game_config.patch_common_this_event.Set(flagValue);
+	if (flagName == "2kbattle")//2k battle Mode
+		lcf::Data::system.easyrpg_use_rpg2k_battle_system = flagValue;
```

Comment:
The strings should be consistent with the command line:

``maniac``, ``dynrpg``,
``key-patch``, ``pic-unlock``, ``common-this``, ``rpg2k3-commands``, (``rpg2k-battle``)

Reply:
Files renamed and reordered to reflect the list of patches.